src/ko/proposal-regexp-named-groups.md

```diff
@@ -0,0 +1,151 @@
+# RegExp Named Capture Groups 번역
+
+4 단계
+
+제안한 사람: Daniel Ehrenberg (Igalia) & Mathias Bynens (Google)
+
+## 서론
+
+번호지정 캡처 그룹을 사용하면 정규표현식과 일치하는 문자열의 특정 부분을 참조할 수 있습니다. 각 캡처 그룹에는 고유한 번호가 할당되어 있으며 이 번호를 사용하여 참조될 수 있지만, 정규표현식을 파악하거나 리팩터링하기 어려울 수 있습니다.
+
+예를 들어 날짜와 일치하는 `/(\d{4})-(\d{2})-(\d{2})/` 를 지정하면 주변 코드를 살펴보지 않고서는 어떤 그룹이 월에 해당하고 어떤 그룹이 일에 해당하는지 알 수 없습니다. 게다가 월과 일의 순서를 바꾸려면 그룹 참조 또한 변경되어야 합니다.
+
+명명된 캡처 그룹은 이러한 이슈들에 대한 훌륭한 해결책을 제공합니다. 
+
+## 고급 API
+
+캡처 그룹에는 모든 식별자 `이름`에 대해 `(?<name>…)` 구문을 사용하여 이름을 지정할 수 있습니다. 날짜에 대한 정규표현식은 `/(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/u` 과 같이 쓰여질 수 있습니다. 각각의 이름은 고유해야 하고 ECMAScript 식별자이름 문법<sup>[1][]</sup>을 따라야 합니다.
+
+명명된 그룹은 정규 표현식 결과의 그룹 속성에서 접근할 수 있습니다. 명명되지 않은 그룹과 마찬가지로 그룹에 대한 번호지정 참조도 생성됩니다. 예를 들어:
+
+```js
+let re = /(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/u; // '/u' : 유니코드 문자에 일치
+let result = re.exec('2015-01-02'); //match 메서드와 유사하지만 첫 번째 매칭 결과만 반환
+// result.groups.year === '2015';
+// result.groups.month === '01';
+// result.groups.day === '02';
+
+// result[0] === '2015-01-02';
+// result[1] === '2015';
+// result[2] === '01';
+// result[3] === '02';
+```
+
+인터페이스는 구조 분해 할당에서도 훌륭하게 동작합니다. 예를 들어:
+
+```js
+let {groups: {one, two}} = /^(?<one>.*):(?<two>.*)$/u.exec('foo:bar'); //여기서 '.'은 모든 문자열을 뜻함 '*'는 없거나 있거나
+console.log(`one: ${one}, two: ${two}`);  // prints one: foo, two: bar
+```
+
+### 역참조
+
+명명된 그룹은 `\k<name>` 구조를 통해 정규표현식 내에서 접근할 수 있습니다. 예를 들어:
+
+```js
+let duplicate = /^(?<half>.*).\k<half>$/u; // '아무문자A''아무문자B''아무문자A'형태
+duplicate.test('a*b'); // false
+duplicate.test('a*a'); // true
+```
+
+명명된 참조는 번호지정 참조와 동시에 사용될 수도 있습니다.
+
+```js
+let triplicate = /^(?<part>.*).\k<part>.\1$/u; 
+triplicate.test('a*a*a'); // true
+triplicate.test('a*a*b'); // false
+```
+
+### 타겟 교체
+
+명명된 그룹은 `String.prototype.replace`에 전달된 대체 값에서도 참조할 수 있습니다. 만약 값이 문자열이라면 `$<name>` 구문을 사용하여 명명된 그룹에 접근할 수 있습니다. 예를 들어: 
+
+```js
+let re = /(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{2})/u;
+let result = '2015-01-02'.replace(re,'$<day>/$<month>/$<year>');
+// result === '02/01/2015'
+```
+
+템플릿 리터럴이 아닌 일반 문자열 리터럴이 `replace`로 전달되었다는 것에 주의하세요. 이 메서드는 `day`등의 값을 로컬 변수로 가지고 있는 것을 대체합니다. 다른 대안으로는 `${day}` 구문을 사용하는 것이 있습니다(템플릿 문자열은 아님). 본 제안서는 `$<day>`를 사용하여 그룹의 정의와 유사하게 그리고 템플릿 리터럴과는 구분되게 했습니다.
```

Comment:
```suggestion
템플릿 리터럴이 아닌 일반 문자열 리터럴이 `replace`로 전달되었다는 것에 주의하세요. 이 메서드는 `day` 등의 값을 지역 변수로 두지 않고 해결하기 때문입니다. 다른 대안으로는 `${day}` 구문을 사용하는 것이 있습니다(템플릿 문자열은 아님). 이 제안은 `$<day>`를 사용하여 그룹의 정의와 유사하게 그리고 템플릿 리터럴과는 구분되게 했습니다.
```

> as that method will resolve the values of day etc rather than having them as local variables. 

이 메서드는 `day` 등의 값을 지역 변수로 두지 않고 해결하기 때문입니다.

> 이 제안은

이 조금 더 자연스러워 보여서 리뷰 드렸습니다.